src/crimson/os/seastore/omap_manager.h

```diff
@@ -97,16 +96,51 @@ class OMapManager {
    *        it it is not set, list all keys after string start.
    * @retval listed key->value mapping and next key
    */
+  struct omap_list_config_t {
+    size_t max_result_size = 128;
+    bool inclusive = false;
+
+    omap_list_config_t(
+      size_t max_result_size,
+      bool inclusive)
+      : max_result_size(max_result_size),
+	inclusive(inclusive) {}
+    omap_list_config_t() {}
+    omap_list_config_t(const omap_list_config_t &) = default;
+    omap_list_config_t(omap_list_config_t &&) = default;
+    omap_list_config_t &operator=(const omap_list_config_t &) = default;
+    omap_list_config_t &operator=(omap_list_config_t &&) = default;
+
+    static omap_list_config_t with_max(size_t max) {
+      omap_list_config_t ret{};
+      ret.max_result_size = max;
+      return ret;
+    }
+
+    static omap_list_config_t with_inclusive(bool inclusive) {
+      omap_list_config_t ret{};
+      ret.inclusive = inclusive;
+      return ret;
+    }
+
+    auto with_reduced_max(size_t reduced_by) const {
+      assert(reduced_by <= max_result_size);
+      return omap_list_config_t(
+	max_result_size - reduced_by,
+	inclusive
+      );
+    }
+  };
   using omap_list_ertr = base_ertr;
   using omap_list_bare_ret = std::pair<
     bool,
-    std::map<std::string, bufferlist>>;
+    std::map<std::string, bufferlist, std::less<>>>;
```

Comment:
i like this change. because, unlike `std::less<string>`, the transparent comparator does not force us to create a temporary object of `Key` type just for, for instance, looking up in the map. instead, any K would work as long as it can be compared with the `Key`.

Reply:
It also matches the external futurized_store interface.